test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Actions/UIActionEngineTests.cs

```diff
@@ -121,6 +123,59 @@ public async Task GetPreviewResultsAsync_WithMultipleActions_SortsPackageIdentit
             Assert.Equal(packageIdentityC.Id, addedResults[2].Id);
         }
 
+        [Fact]
+        public void AddUiActionEngineTelemetryProperties_AddsVulnerabilityInfo_Succeeds()
+        {
+            // Arrange
+            var telemetrySession = new Mock<ITelemetrySession>();
+            TelemetryEvent lastTelemetryEvent = null;
+            telemetrySession
+                .Setup(x => x.PostEvent(It.IsAny<TelemetryEvent>()))
+                .Callback<TelemetryEvent>(x => lastTelemetryEvent = x);
+
+            var operationId = Guid.NewGuid().ToString();
+
+            var actionTelemetryData = new VSActionsTelemetryEvent(
+                operationId,
+                projectIds: new[] { Guid.NewGuid().ToString() },
+                operationType: NuGetOperationType.Install,
+                source: OperationSource.PMC,
+                startTime: DateTimeOffset.Now.AddSeconds(-1),
+                status: NuGetOperationStatus.NoOp,
+                packageCount: 1,
+                endTime: DateTimeOffset.Now,
+                duration: .40);
+
+            UIActionEngine.AddUiActionEngineTelemetryProperties(
+                actionTelemetryEvent: actionTelemetryData,
+                continueAfterPreview: true,
+                acceptedLicense: true,
+                userAction: UserAction.CreateInstallAction("mypackageId", new NuGetVersion(1, 0, 0)),
+                selectedIndex: 0,
+                recommendedCount: 0,
+                recommendPackages: false,
+                recommenderVersion: null,
+                topLevelVulnerablePackagesCount: 1,
+                topLevelVulnerablePackagesMaxSeverities: new List<int> { 1, 1, 3  },
+                existingPackages: null,
+                addedPackages: null,
+                removedPackages: null,
+                updatedPackagesOld: null,
+                updatedPackagesNew: null,
+                targetFrameworks: null);
+
+            // Act
+            var service = new NuGetVSTelemetryService(telemetrySession.Object);
+            service.EmitTelemetryEvent(actionTelemetryData);
+
+            // Assert
+            Assert.NotNull(lastTelemetryEvent);
+            Assert.Equal(1, lastTelemetryEvent["TopLevelVulnerablePackagesCount"]);
+            Assert.NotNull(lastTelemetryEvent.ComplexData["TopLevelVulnerablePackagesMaxSeverities"] as List<int>);
+            var pkgSeverities = lastTelemetryEvent.ComplexData["TopLevelVulnerablePackagesMaxSeverities"] as List<int>;
+            Assert.Equal(3, pkgSeverities.Count());
```

Comment:
Assert that it contains 1, 1, 3? 
If you change the sort per my other comment, then assert the order as well.

Reply:
Added collection assertion.